### PR TITLE
Improve ram_group_membership resource use common method

### PR DESCRIPTION
**Pull request by Lexsss**

### CODE REVIEW

File: alicloud/resource_alicloud_ram_group_membership_test.go [hunk 1]
```diff
@@ -12,67 +12,142 @@ import (
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheck(groupMenbershipMap),
 				),
 			},
 		},
 	})
+}
 
+var groupMenbershipMap = map[string]string{
+	"group_name":   CHECKSET,
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_ram_group_membership_test.go]: 去掉 group_name

File: alicloud/service_alicloud_ram.go [hunk 2]
```diff
@@ -246,6 +247,52 @@ func (s *RamService) WaitForRamUser(id string, status Status, timeout int) error
...
+	})
+	if err != nil {
+		if RamEntityNotExist(err) {
+			return nil, WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
+		}
+		return nil, WrapErrorf(err, DefaultErrorMsg, id, request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
+	response = raw.(*ram.ListUsersForGroupResponse)
+	return
```
Comment (ddaddaddda) [inline alicloud/service_alicloud_ram.go]: 加 if id !=id{...} 判断

File: alicloud/service_alicloud_ram.go [hunk 3]
```diff
@@ -246,6 +247,52 @@ func (s *RamService) WaitForRamUser(id string, status Status, timeout int) error
...
+		if object != nil {
+			if len(object.Users.User) == 0 {
+				if status == Deleted {
+					return nil
+				}
+			} else {
+				if status != Deleted {
+					return nil
+				}
+			}
```
Comment (ddaddaddda) [inline alicloud/service_alicloud_ram.go]: 281-289    我觉得等效于：
if status == Deleted {
     return nil
}

File: alicloud/service_alicloud_ram.go [hunk 4]
```diff
@@ -246,6 +247,52 @@ func (s *RamService) WaitForRamUser(id string, status Status, timeout int) error
...
+			if NotFoundError(err) {
+				if status == Deleted {
+					return nil
+				}
+			} else {
+				return WrapError(err)
+			}
+		}
+		if object != nil {
+			if len(object.Users.User) == 0 {
```
Comment (xiaozhu36) [inline alicloud/service_alicloud_ram.go]: still applies

**Pull request merged**